# Project X -- [Prisha ] [Joshi]

**TA Help:** John Smith, Alice Jones

- Help with figuring out how to write a function.
    
**Collaboration:** Friend1, Friend2
    
- Helped figuring out how to load the dataset.
- Helped debug error with my plot.

## Question 1

In [6]:
library(data.table)
productsDF <- fread("/anvil/projects/tdm/data/icecream/combined/products.csv")
reviewsDF <- fread("/anvil/projects/tdm/data/icecream/combined/reviews.csv")
newmergedDF <- merge(productsDF, reviewsDF, by = c("brand", "key") )
ChocolateChipCookieDoughDF <- subset(newmergedDF, (brand == "bj") & (key == "16_bj"))
meandf <- mean(ChocolateChipCookieDoughDF$stars, na.rm= TRUE)
meandf

[1] 4.560327

the mean star rating for chocolatechip cookie dough icecream is 4.5603. 

## Question 2

In [20]:
HalfBakedDF <- subset(newmergedDF,(name == "Half Baked\302\256"))
starsdf <- mean(HalfBakedDF$stars)
starsdf



[1] 4.695701

the average number of stars for the half baked ice cream is 4.6957. 

## Question 3

In [38]:
chocolatedf <-  newmergedDF[grepl("Chocolate",newmergedDF$name)]
averagedf<- mean(chocolatedf$stars)
averagedf

[1] 4.154833

the average number of stars for any ice cream that had the word "chocolate" in it is 4.1548.

## Question 4

In [2]:
library(data.table)
airportsDF <- fread("/anvil/projects/tdm/data/flights/subset/airports.csv")
flightdataDF <- fread("/anvil/projects/tdm/data/flights/subset/1990.csv")
mybigDF <- merge(airportsDF, flightdataDF, by.x = "iata", by.y = "Origin")
newdf <-mean(mybigDF$DepDelay[mybigDF$iata == "IND"],na.rm=TRUE)
newdf
length(mybigDF$Origin )

Warning message in fread("/anvil/projects/tdm/data/flights/subset/airports.csv"):
"Found and resolved improper quoting out-of-sample. First healed line 1253: <<"DBN","W. H. \"Bud\" Barron ","Dublin","GA","USA",32.56445806,-82.98525556>>. If the fields are not quoted (e.g. field separator does not appear within any field), try quote="" to avoid this warning."


[1] 5.969772

[1] 0

the average delay from the origin given us nan. this could mean that the merged df does not have origin as a column in the new df. even the function length(mybigDF$Origin) gives us 0 which might indicate the same thing. using ind as the iata code gives us the correct answer which is 5.9698. 

## Question 5

In [14]:
newdf2 <-mean(mybigDF$DepDelay[mybigDF$iata == "HOU"],na.rm=TRUE)
newdf2
newdf3 <-mean(mybigDF$DepDelay[mybigDF$iata == "IAH"],na.rm=TRUE)
newdf3

[1] 8.446508

[1] 6.475591

since there are two airports in houston texas ,their departure delays from different airport codes are shown above. 

## Pledge

By submitting this work I hereby pledge that this is my own, personal work. I've acknowledged in the designated place at the top of this file all sources that I used to complete said work, including but not limited to: online resources, books, and electronic communications. I've noted all collaboration with fellow students and/or TA's. I did not copy or plagiarize another's work.

> As a Boilermaker pursuing academic excellence, I pledge to be honest and true in all that I do. Accountable together – We are Purdue.